dataSourceImpl/src/main/java/com/panhuk/dataSourceimpl/DatastorePreferencesImpl.kt

```diff
@@ -38,18 +38,18 @@ class DatastorePreferencesImpl(private val dataStore: DataStore<Preferences>) :
     }
   }
 
-  override fun isFirstTime(): Flow<Boolean> {
+  override fun isFirstTimeAppOpened(): Flow<Boolean> {
```

Comment:
how about splitting the logic into two methods: one just checks and the other one sets the value

right now this is not obvious that simply checking if app was opened before will also change the value

Reply:
Well, I set a new method in interface to change the value to false but is it good to add this method to interface. I am not sure that I've made it right